src/ctype/CTypeMetadata.ts

```diff
@@ -0,0 +1,17 @@
+import { ICtypeMetadata } from '../types/CType'
+import CType from './CType'
+import { CTypeWrapperMetadata } from './CTypeSchema'
+import * as CTypeUtils from './CTypeUtils'
+
+export default class CTypeMetadata implements ICtypeMetadata {
+  public ctypeHash: ICtypeMetadata['ctypeHash']
+  public metadata: ICtypeMetadata['metadata']
+
+  public constructor(ctype: CType, ctypeMetadata: CTypeMetadata) {
+    if (!CTypeUtils.verifySchema(ctypeMetadata, CTypeWrapperMetadata)) {
+      throw new Error('CTypeMetadata does not correspond to schema')
+    }
+    this.metadata = ctypeMetadata.metadata
+    this.ctypeHash = ctype.hash
+  }
+}
```

Comment:
I am quite happy with this, it is clean and to the point.

Only thing is the CType, as I feel I have too extra information.

Reply:
I have updated the constructor to take the `hash`